centreon/src/Core/AdditionalConnectorConfiguration/Domain/Model/VmWareV6Parameters.php

```diff
@@ -82,6 +82,56 @@ public function __construct(
     }
 
     /**
+     * @inheritDoc
+     *
+     * @param EncryptionInterface $encryption
+     * @param AccParametersInterface $currentObj
+     * @param array<string,mixed> $newDatas
+     *
+     * @return VmWareV6Parameters
+     */
+    public static function update(
+        EncryptionInterface $encryption,
+        AccParametersInterface $currentObj,
+        array $newDatas
+    ): self
+    {
+        /** @var _VmWareV6Parameters|_VmWareV6ParametersWithoutCredentials $newDatas */
+        /** @var _VmWareV6Parameters $parameters */
+        $parameters = $currentObj->getDecryptedData();
+
+        foreach ($newDatas['vcenters'] as $index => $vcenter) {
+            $newDatas['vcenters'][$vcenter['name']] = $vcenter;
+            unset($newDatas['vcenters'][$index]);
+        }
+
+        $parameters['port'] = $newDatas['port'];
+        foreach ($parameters['vcenters'] as $index => $vcenter) {
+            if (! array_key_exists($vcenter['name'], $newDatas['vcenters'])) {
+                unset($parameters['vcenters'][$index]);
+
+                continue;
+            }
+            $updatedVcenter = $newDatas['vcenters'][$vcenter['name']];
+            $updatedVcenter['username'] ??= $vcenter['username'];
+            $updatedVcenter['password'] ??= $vcenter['password'];
+
+            $parameters['vcenters'][$index] = $updatedVcenter;
+            unset($newDatas['vcenters'][$vcenter['name']]);
+        }
+        if ([] !== $newDatas['vcenters']) {
+            foreach ($newDatas['vcenters'] as $newVcenter) {
+                $parameters['vcenters'][] = $newVcenter;
+            }
+        }
+        $parameters['vcenters'] = array_values($parameters['vcenters']);
```

Comment:
Maybe add a comment to explain what is done here, it's quiet difficult to understand while reading